core/src/main/java/io/questdb/griffin/engine/functions/IntFunction.java

```diff
@@ -139,7 +139,15 @@ public RecordCursorFactory getRecordCursorFactory() {
 
     @Override
     public final short getShort(Record rec) {
-        throw new UnsupportedOperationException();
+        final int value = getInt(rec);
```

Comment:
we must not be calling `getShort` on an `int` function. Lets discuss why this change is here